services/apps/alcs/src/portal/public/search/application/public-application-search.service.ts

```diff
@@ -0,0 +1,305 @@
+      //     },
+      //   );
+      // }
+      //
+      // if (searchDto.resolutionYear !== undefined) {
+      //   query = query.andWhere('decision.resolution_year = :resolution_year', {
+      //     resolution_year: searchDto.resolutionYear,
+      //   });
+      // }
+    }
+    return query;
+  }
+
+  private joinApplicationDecision(query: any) {
+    query = query.leftJoin(
+      ApplicationDecision,
+      'decision',
+      'decision.application_uuid = "appSearch"."application_uuid"',
+    );
+    return query;
+  }
+
+  private compileApplicationParcelSearchQuery(
+    searchDto: SearchRequestDto,
+    query,
+  ) {
+    if (searchDto.pid || searchDto.civicAddress) {
+      query = query.leftJoin(
+        ApplicationParcel,
+        'parcel',
+        "parcel.application_submission_uuid = appSearch.uuid AND parcel.parcel_type IN ('application', 'other')",
+      );
+    } else {
+      query = query.leftJoin(
+        ApplicationParcel,
+        'parcel',
+        "parcel.application_submission_uuid = appSearch.uuid AND parcel.parcel_type = 'application'",
+      );
+    }
+
+    if (searchDto.pid) {
+      query = query.andWhere('parcel.pid = :pid', { pid: searchDto.pid });
+    }
+
+    if (searchDto.civicAddress) {
+      query = query.andWhere('parcel.civic_address like :civic_address', {
+        civic_address: `%${searchDto.civicAddress}%`,
+      });
+    }
+    return query;
+  }
+
+  private compileApplicationSearchByNameQuery(
+    searchDto: SearchRequestDto,
+    query,
+  ) {
+    if (searchDto.name) {
+      const formattedSearchString =
+        formatStringToPostgresSearchStringArrayWithWildCard(searchDto.name!);
+
+      query = query
+        .leftJoin(
+          ApplicationOwner,
+          'application_owner',
+          'application_owner.application_submission_uuid = appSearch.uuid',
+        )
+        .andWhere(
+          new Brackets((qb) =>
+            qb
+              .where(
+                "LOWER(application_owner.first_name || ' ' || application_owner.last_name) LIKE ANY (:names)",
+                {
+                  names: formattedSearchString,
+                },
+              )
+              .orWhere(
+                'LOWER(application_owner.first_name) LIKE ANY (:names)',
+                {
+                  names: formattedSearchString,
+                },
+              )
+              .orWhere('LOWER(application_owner.last_name) LIKE ANY (:names)', {
+                names: formattedSearchString,
+              })
+              .orWhere(
+                'LOWER(application_owner.organization_name) LIKE ANY (:names)',
+                {
+                  names: formattedSearchString,
+                },
+              ),
+          ),
+        );
+    }
+    return query;
+  }
+
+  private compileApplicationFileTypeSearchQuery(
+    searchDto: SearchRequestDto,
+    query,
+  ) {
+    if (searchDto.fileTypes.length > 0) {
```

Comment:
do you need to adjust this code according to the commented out decision code above?

Reply:
Maybe, I'm not sure yet